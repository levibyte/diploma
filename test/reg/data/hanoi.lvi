function Hanoi(var m, var a, var b, var c){
  moves++;
  if(m == 1){
   print << m << " from " << a << " to " << c << endl;
  }else{  
    Hanoi(m-1, a,c,b);
    cout << "Move disc " << m << " from " << a << " to " << c << endl;
    Hanoi(m-1,b,a,c);
  }
}

function main(){
  var discs = 4;
  var result = Hanoi(discs, 'A', 'B', 'C');
  print result;
}
