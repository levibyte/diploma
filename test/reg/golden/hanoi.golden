Expect ';' after expression.
Expect ';' after variable declaration.
Expect ';' after variable declaration.
Expect ';' after variable declaration.
Expect ';' after variable declaration.
expect expression.
Expect ';' after expression.
expect expression.
Expect ';' after expression.
expect expression.
Expect ';' after variable declaration.
expect expression.
Undefined variable m.
Operands must be number
